pkg/storage/stores/indexshipper/compactor/compactor.go

```diff
@@ -401,9 +443,114 @@ func (c *Compactor) starting(ctx context.Context) (err error) {
 	}
 	level.Info(util_log.Logger).Log("msg", "compactor is ACTIVE in the ring")
 
+	if err != nil {
+		return err
+	}
+
 	return nil
 }
 
+func (c *Compactor) sizeBasedCompactionInterval(ctx context.Context) error {
+	if exceeded, err := c.sizeBasedRetention.ThresholdExceeded(); !exceeded {
```

Comment:
This doesn't handle the error when `exceeded == true`

Reply:
Hi @MasslessParticle 

We are not handling this situation, because [the function `ThresholdExceeded()`](
https://github.com/grafana/loki/pull/7927/files#diff-8357ce3aacb6cace0adbbbb88de67c6808980aa2e7859697355571864b5057ddR124-R134) does not return this combination: `true, err`. The possible returns are:

- `false, err`
- `false, nil`
- `true, nil`

Do you think we should modify this anyway?

Reply:
For uniform error handling, I think it would be best to modify this to handle `true, err`. This guards against future changes to `ThresholdExceeded`

Reply:
Reviewing the code again seems that there is no point in checking all the combinations. 

We need just 2 guards:

- Log a message and `return nil` if there is an error, or
- `return nil` if doesn't exceed threshold.

In other situations we should continue with the execution.

Pushing these changes with less nesting, let me know your thoughts.

Reply:
This makes sense. Do we return `nil` on error to not exit compactions? That makes sense, but we should probably add a metric here so an alert can be raised on successive failures.